src/packages/rescue/rat_rescue.py

```diff
@@ -783,27 +785,22 @@ def mark_delete(self, reporter: str, reason: str) -> None:
         Raises:
             TypeError: invalid params
         """
-        # type enforcement
-        if not isinstance(reporter, str) or not isinstance(reason, str):
-            raise TypeError(f"reporter and/or reason of invalid type. got {type(reporter)},"
-                            f"{type(reason)}")
+
+        self.marked_for_deletion = MarkForDeletion(reporter=reporter, reason=reason, marked=True)
+        if (reporter or reason) and (not reporter or not reason):
```

Comment:
[Question] Why are we validating the inputs AFTER we have already used them to create the MFD?

Reply:
MFD only validates the object is strictly invalid, I reasoned it was possible that strictly valid MFD objects may exist that fail this extra check, but mecha should refuse to create new instances.

Reply:
I'm not sure I understand.
I can see that `MarkForDeletion` has a validator to ensure that `reporter` and `reason` cannot be an empty string, but they're also optional, meaning they can be `None`.
This check does absolutely nothing to prevent creating a new `MarkForDeletion` instance with `reason` and `reporter` both set to `None`, as the object is instantiated and assigned to the `marked_for_deletion` member immediately. If this check fails and decides to raise, the only thing that ends up being skipped is the additional log call below.

Reply:
the MFD fields can be null, this is intended behavor.

the validators on MFD only validate that - if its not null - the attributes are non-empty strings.

Reply:
> This check ensures that , if MFD is filled in, its filled in entirely. you can't MFD without a reason.

Mmm, but that's not true either.

So, imagine for instance we call `rescue.mark_delete("Eearslya", None)`.
On line 789, the MFD object is created, instantiated, and assigned to the rescue with `reporter = "Eearslya"` and `reason = None`.

At this point in time, **the rescue is marked for deletion**.

On line 790, we check to make sure that if one field is filled out, they both are. Because `reason` is `None`, this check will raise an error.
But, the problem is...the MFD object is already attached to the rescue. There's no code that removes it if it's found to be invalid.

Reply:
... i could have sworn i implemented this so the assignment occurs after 🤦‍♂ 
Further, the tests for this routine didn't cover this case properly. fix incoming.

Reply:
9ba7b9ef